tests/typ/text/hyphenate.typ

```diff
@@ -49,3 +49,59 @@ It's a #emph[Tree]beard.
 #set page(width: 60pt)
 #set text(hyphenate: true)
 #h(6pt) networks, the rest.
+
+---
+#set page(height: 60pt)
+
+#let sample = lorem(12)
+
+#sample
+#pagebreak()
+#set text(costs: (widow: 0%, orphan: 0%))
+#sample
+
+---
+#set par(justify: true)
+
+#let sample = [please avoid runts in this text.]
+
+#sample
+#pagebreak()
+#set text(costs: (runt: 10000%))
+#sample
+
+---
+#set par(justify: true)
+#set text(size: 6pt)
+
+#let sample = [a a a a a a a a a a a a a a a a a a a a a a a a a]
+
+#sample
+#pagebreak()
+#set text(costs: (runt: 0%))
+#sample
+
+---
+#set par(justify: true)
+
+#let sample = [we've increase the hyphenation cost.]
```

Comment:
I assume the grammar mistake is necessary to get the desired hyphenation choices? ^^

Reply:
I thought it was, but now it seems that it's not.